packages/core/src/install/index.ts

```diff
@@ -1004,7 +1004,7 @@ const _installInContext: InstallFunction = async (projects, ctx, opts) => {
       const projectToInstall = projects[index]
       if (opts.global && projectToInstall.mutation.includes('install')) {
         projectToInstall.wantedDependencies.forEach(pkg => {
-          if (!linkedPackages?.includes(pkg.alias)) {
+          if (pkg.alias && !linkedPackages?.includes(pkg.alias)) {
             logger.warn({ message: `${pkg.alias} has no binaries`, prefix: opts.lockfileDir })
```

Comment:
but there is still an issue, no? Maybe something else should be printed instead of the alias.

Reply:
Yes, but the warning `undefined has no binaries` is more confusing.

Reply:
I mean the pkg has other fields that may be printed instead of the alias
```suggestion
          if (!linkedPackages?.includes(pkg.alias)) {
            logger.warn({ message: `${pkg.alias ?? pkg.pref} has no binaries`, prefix: opts.lockfileDir })
```

Reply:
Yeah, I found it's hard to parse alias of `link:xxx` in `parseWantedDependencies`, so it's better to use another field